### PR TITLE
feat(eks): support for Kubernetes version 1.22

### LINKED ISSUE #20263: (eks): support latest EKS engine version v1.22
### Describe the bug

We just updated the CDK version from 2.20 to 2.23 and we are using the latest EKS 1.22 version. Looks like the EKS 1.22 is removed from 2.22+. 

Error from npm build: "error TS2551: Property 'V1_22' does not exist on type 'typeof KubernetesVersion'. Did you mean 'V1_20'?"

### Expected Behavior

EKS v1.22 should not be removed.

### Current Behavior

EKS v1.22 is not supported.

### Reproduction Steps

Specify Kubernete version of V1_22 when create an EKS cluster using CDK 2.22+.

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.23

### Framework Version

_No response_

### Node.js Version

14

### OS

Mac

### Language

Typescript

### Language Version

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by madeline-k**
Add support for Kubernetes Version 1.22. In order to use this version, customers must pass in a `KubectlLayer` object from `@aws-cdk/lambda-layer-kubectl-v22` to the `kubectlLayer` construct prop of `Cluster`. 

Closes #20263 

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by MrArnoldPalmer — CHANGES_REQUESTED** [review]
I would add a section to the readme [here](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_eks-readme.html#kubectl-support) about providing custom kubectl CLI versions. I think important things to call out are.

1. The version of kubectl provided must be compatible with the Kubernetes version that the cluster is using.
2. A basic code example of importing the kubectl asset package and passing to the appropriate prop.

File: packages/@aws-cdk/aws-eks/package.json [hunk 1]
```diff
@@ -80,6 +80,7 @@
   },
   "license": "Apache-2.0",
   "devDependencies": {
+    "@aws-cdk/lambda-layer-kubectl-v22": "0.0.4",
```
Comment (corymhall) [inline packages/@aws-cdk/aws-eks/package.json:83]: Should this depend on `v2.0.0`? I'm assuming that is the stable version?

Comment (madeline-k) [reply packages/@aws-cdk/aws-eks/package.json:83]: Yep you are right, was waiting for that to get released.

**Review by MrArnoldPalmer — APPROVED** [review]
🎉

**Comment by henrysachs** [comment]
can we include 1.23 also in this pr? I think #22544 tried something like that but your PR is more "complete".

**Comment by madeline-k** [comment]
> can we include 1.23 also in this pr? I think #22544 tried something like that but your PR is more "complete".

@henrysachs , we are planning to do 1.23 and 1.24 in the same manner as this. But, those will come in subsequent PRs.

**Pull request merged**